net-p2p/bitcoin-core/bitcoin-core-25.0.ebuild

```diff
@@ -0,0 +1,251 @@
+src_prepare() {
+	default
+	! use system-leveldb || rm -r src/leveldb || die
+	if use system-libsecp256k1 ; then
+		rm -r src/secp256k1 || die
+		sed -e '/^DIST_SUBDIRS *=/s/\bsecp256k1\b//' -i src/Makefile.am || die
+	else
+		pushd src/secp256k1 >/dev/null || die
+		AT_NOELIBTOOLIZE=yes eautoreconf
+		popd >/dev/null || die
+	fi
+	eautoreconf
+
+	# we say --disable-util-util, so we can't test bitcoin-util
+	sed -ne '/^  {/{h;:0;n;H;/^  }/!b0;g;\|"exec": *"\./bitcoin-util"|d};p' \
+		-i test/util/data/bitcoin-util-test.json || die
+
+	sed -e 's/^\(complete -F _bitcoind\b\).*$/\1'"$(usev daemon ' bitcoind')$(usev qt5 ' bitcoin-qt')/" \
+		-i contrib/completions/bash/bitcoind.bash-completion || die
+}
+
+src_configure() {
+	local wallet ; if use berkdb || use sqlite ; then wallet=enable ; else wallet=disable ; fi
+	local myeconfargs=(
+		--disable-static
+		--${wallet}-wallet
+		$(use_with sqlite)
+		$(use_with berkdb bdb)
+		$(use_enable systemtap usdt)
+		$(use_with upnp miniupnpc)
+		$(use_with nat-pmp natpmp)
+		$(use_enable test tests)
+		--disable-bench
+		--disable-fuzz{,-binary}
+		$(use_with qrcode qrencode)
+		--disable-ccache
+		$(use_enable asm)
+		$(use_enable zeromq zmq)
+		$(use_enable man)
+		$(use_enable external-signer)
+		--with-utils
+		$(use_enable {bitcoin,util}-cli)
+		--enable-util-tx
+		--${wallet}-util-wallet
+		--disable-util-util
+		$(use_with libs)
+		$(use_with daemon)
+		$(use_with qt5 gui qt5)
+		$(use_with dbus qtdbus)
+		$(use_with system-leveldb)
+		$(use_with system-libsecp256k1)
+	)
+	econf "${myeconfargs[@]}"
+}
+
+src_install() {
+	use external-signer && DOCS+=( doc/external-signer.md )
+	use berkdb || use sqlite && DOCS+=( doc/managing-wallets.md )
+	use libs && DOCS+=( doc/shared-libraries.md )
+	use systemtap && DOCS+=( doc/tracing.md )
+	use zeromq && DOCS+=( doc/zmq.md )
+
+	default
+
+	! use libs || find "${ED}" -name '*.la' -delete || die
+	! use test || rm -f -- "${ED}"/usr/bin/test_bitcoin{,-qt} || die
+
+	newbashcomp contrib/completions/bash/bitcoin-tx.bash-completion bitcoin-tx
+	use bitcoin-cli && newbashcomp contrib/completions/bash/bitcoin-cli.bash-completion bitcoin-cli
+	use daemon || use qt5 && newbashcomp contrib/completions/bash/bitcoind.bash-completion bitcoind
+
+	if use daemon ; then
+		insinto /etc/bitcoin
+		sed -ne 's/ To use, copy this file$//p;Tp;:0;n;/save the file\.$/!b0;d;:p;p' \
+			share/examples/bitcoin.conf >"${ED}/etc/bitcoin/bitcoin.conf" || die
+		fowners bitcoin:bitcoin /etc/bitcoin/bitcoin.conf
+		fperms 600 /etc/bitcoin/bitcoin.conf
+
+		newconfd contrib/init/bitcoind.openrcconf bitcoind
+		newinitd "${FILESDIR}/bitcoind.openrc" bitcoind
+		systemd_newunit contrib/init/bitcoind.service bitcoind.service
+
+		keepdir /var/lib/bitcoind
+		fperms 700 /var/lib/bitcoind
+		fowners bitcoin:bitcoin /var/lib/bitcoind
+		dosym -r /etc/bitcoin/bitcoin.conf /var/lib/bitcoind/bitcoin.conf
+		dosym -r /var/lib/bitcoind /var/lib/bitcoin/.bitcoin
+
+		doman "${FILESDIR}/bitcoin.conf.5"
```

Comment:
Personal opinion: I would drop the man page. This is upstream's job. Plus, this man page was taken from Debian, which removed its bitcoin packages, so one could say that the man page is not maintained. Unless you want to maintain it. But even then, it may be better to maintain it upstream.

Reply:
I'm perfectly happy to. I only included it to avoid any regressions versus the status quo. If I had omitted it, someone would have complained, and I would have been less likely to get my PR merged.